Util.xs

```diff
@@ -57,17 +57,56 @@
-        OpLASTSIB_set(arg, argparent);                                      \
+                                                                            \
+        /* extract the args for the custom op, and delete the remaining ops \
+           NOTE: this is the *single* arg version, multi-arg is more        \
+           complicated, see Hash::SharedMem's THX_ck_entersub_args_hsm */   \
+                                                                            \
+        /* These comments will visualize how the op tree look like after    \
+           each operation. We usually start out with this: */               \
+        /*** entersub( list( push, arg1, cv ) ) */                          \
+        /* Though in rare cases it can also look like this: */              \
+        /*** entersub( push, arg1, cv ) */                                  \
+                                                                            \
+        /* first, get the real pushop, after which comes the arg list */    \
+        OP *pushop = cUNOPx( entersubop )->op_first;    /* Cast the entersub op as an op with a single child            \
+                                                           and get that child (the args list or pushop). */             \
+        if( !OpHAS_SIBLING( pushop ) )          /* Go one layer deeper to get at the real pushop. */                    \
+          pushop = cUNOPx( pushop )->op_first;  /* (Sometimes not necessary when pushop is directly on entersub.) */    \
+                                                                            \
+        /* then extract the arg */                                          \
+        OP *arg = OpSIBLING( pushop );  /* Get a pointer to the first arg op                                            \
+                                           so we can attach it to the custom op later on. */                            \
+        /*** entersub( list( push, arg1, cv ) ) + ( arg1, cv ) */           \
+                                                                            \
+        /* and prepare to delete the other ops */                           \
+        OpMORESIB_set( pushop, OpSIBLING( arg ) );  /* Replace the first op of the arg list with the last arg op        \
+                                                       (the cv op, i.e. pointer to original xs function),               \
+                                                       which allows recursive deletion of all unneeded ops              \
+                                                       while keeping the arg list. */                                   \
+        /*** entersub( list( push, cv ) ) + ( arg1, cv ) */                 \
+                                                                            \
+        OpLASTSIB_set( arg, NULL ); /* Remove the trailing cv op from the arg list,                                     \
```

Comment:
After looking at the source code:

    #  define OpLASTSIB_set(o, parent) \
           ((o)->op_moresib = 0, (o)->op_sibparent = (parent)) 

And assigning `NULL` to `op_sibparent` is fine, since we're cutting it out of the tree and it has no parent anymore.